diff --git a/node_modules/react-native-worklets/cpp/JsiWorkletContext.cpp b/node_modules/react-native-worklets/cpp/JsiWorkletContext.cpp
index b2ea105..2239a97 100644
--- a/node_modules/react-native-worklets/cpp/JsiWorkletContext.cpp
+++ b/node_modules/react-native-worklets/cpp/JsiWorkletContext.cpp
@@ -119,7 +119,7 @@ void JsiWorkletContext::addDecorator(
     std::shared_ptr<JsiBaseDecorator> decorator) {
   decorators.push_back(decorator);
   // decorate default context
-  JsiWorkletContext::getInstance()->decorate(decorator);
+  // JsiWorkletContext::getInstance()->decorate(decorator);
 }
 
 template <typename... Args> void JsiWorkletContext::decorate(Args &&...args) {
